src/form/input/Input.tsx

```diff
@@ -161,18 +179,64 @@ function Input(props: InputProps) {
 
       if (newValue) {
         const formattedNewValue = parseNumber({locale, maximumFractionDigits}, newValue);
-        const isFormattedNewValueNotAValidNumber = Number.isNaN(
-          Number(formattedNewValue)
-        );
+        // Number("-") returns NaN. Should allow minus sign as first character.
+        const isFormattedNewValueNotAValidNumber =
+          (newValue !== "-" && Number.isNaN(Number(formattedNewValue))) ||
+          formattedNewValue === "";
         let finalEventValue = formattedNewValue ? String(formattedNewValue) : "";
 
-        // IF the parsed number is a valid and there is a decimal separator, we need to save the number as it is so that decimal part doesn't disappear
+        // IF the parsed number is a valid and there is a decimal separator,
+        // we need to save the number as it is so that decimal part doesn't disappear
         if (!isFormattedNewValueNotAValidNumber && formattedNewValue.match(/./)?.length) {
           finalEventValue = String(formattedNewValue);
         } else if (isFormattedNewValueNotAValidNumber) {
+          finalEventValue = value as string | "";
+        }
+
+        // IF 'shouldFormatToLocaleString' or 'maximumFractionDigits' are defined or the value is negative,
+        // value can't have leading zeros. Like 0,000,123 or 010.50 or -00
+        if (
+          !isFormattedNewValueNotAValidNumber &&
+          (shouldFormatToLocaleString ||
+            maximumFractionDigits > 0 ||
+            finalEventValue.includes("-"))
+        ) {
+          finalEventValue = removeLeadingZeros(locale, finalEventValue);
+        }
+
+        // IF maximumFractionDigits is set as 0, value can not be negative zero
+        if (maximumFractionDigits === 0 && finalEventValue === "-0") {
           finalEventValue = value as string;
         }
 
+        // IF shouldFormatToLocaleString is defined, caret position should calculate according to thoudsandths separator count
+        if (shouldFormatToLocaleString) {
+          const numberFormatter = formatNumber({
+            maximumFractionDigits,
+            locale: "en"
+          });
+          const thousandthsSeparatorCount =
+            numberFormatter(parseFloat(finalEventValue)).match(/,/g)?.length || 0;
```

Comment:
Should we have a util for this?
```suggestion
            getThousandthSeparatorCount(finalEventValue)
```

Reply:
Yes, it would be better. I'll add now.